utils/oscap.8

```diff
@@ -82,12 +82,12 @@ Select a particular rule from XCCDF document. Only this rule will be evaluated.
 .TP
 \fB\-\-tailoring-file TAILORING_FILE\fR
 .RS
-Use given file for XCCDF tailoring. If both --tailoring-file and --tailoring-id are specified, --tailoring-file takes priority!
+Use given file for XCCDF tailoring. Needs to specify customized profile ID using --profile to take effect. If both --tailoring-file and --tailoring-id are specified, --tailoring-file takes priority.
```

Comment:
> Needs to specify customized profile ID using --profile to take effect.

You don't have to if the tailoring shadows an existing profile.

Reply:
@mpreisler I wanted to document the situation when it doesn't shadow. People usually think that if they add `--tailoring-file` they're done and then they wonder why the profile isn't tailored. This isn't documented anywhere.

But if it shadows, it means the customized profile ID is the same as original profile ID, so you still can say that you need to specify the customized profile ID using `--profile`, so from this pow the sentence is correct.

I'll try to find better wording.

Reply:
What about instead of  this we suggest user to ensure that he doesn't use the original profile if he wants the tailored profile, an that their IDs can differ?

Reply:
@mpreisler @dahaic What do you think?

Reply:
Others opinions are also valuable :-) @mpreisler @matusmarhefka @matejak @mpreisler @dahaic @WesleyCeraso @yuumasato @redhatrises and anybody else, feel free to comment :)

Reply:
> What about instead of this we suggest user to ensure that he doesn't use the original profile if he wants the tailored profile, an that their IDs can differ?

We can suggest users to not use the same ID for customized Profiles, but I guess we cannot stop them from doing it, I don't know what use cases people have.

Something I noticed now, is that SCAP Workbench 1.1.5 adds `2` to end of Profile ID when provided identical ID to the one being customized. So users shadowing Profiles may be doing it by hand?

> Needs to specify customized profile ID using --profile to take effect.

Suggestion: Needs to specify Profile to apply with --profile.

And can we change of option `--profile` of `xccdf generate` from the help to be like the one in manual page? I find the description in man page much more clear.

I don't understand how to use `--tailoring-id`. Couldn't figure out from manual or help. Tried following and error message didn't help me.
```
[coruscant git]$ oscap xccdf generate fix --output fix.sh --tailoring-id scap_org.open-scap_comp_ssg-rhel7-xccdf-1.2.xml --benchmark-id xccdf_org.ssgproject.content_benchmark_RHEL-7 scap-security-guide/build/ssg-rhel7-ds.xml 
OpenSCAP Error: No 'scap_org.open-scap_comp_ssg-rhel7-xccdf-1.2.xml' component ref found in file 'scap-security-guide/build/ssg-rhel7-ds.xml' in datastream of id 'scap_org.open-scap_datastream_from_xccdf_ssg-rhel7-xccdf-1.2.xml'. [ds_sds_session.c:316]
Could not extract scap_org.open-scap_comp_ssg-rhel7-xccdf-1.2.xml with all dependencies from datastream. [ds_sds_session.c:224]
[coruscant git]$ oscap xccdf generate fix --output fix.sh --tailoring-id xccdf_org.ssgproject.content_profile_standard --benchmark-id xccdf_org.ssgproject.content_benchmark_RHEL-7 scap-security-guide/build/ssg-rhel7-ds.xml 
OpenSCAP Error: No 'xccdf_org.ssgproject.content_profile_standard' component ref found in file 'scap-security-guide/build/ssg-rhel7-ds.xml' in datastream of id 'scap_org.open-scap_datastream_from_xccdf_ssg-rhel7-xccdf-1.2.xml'. [ds_sds_session.c:316]
Could not extract xccdf_org.ssgproject.content_profile_standard with all dependencies from datastream. [ds_sds_session.c:224]
```

Plese clarify. I don't want to look at the source to get spoiled, :P

Reply:
> We can suggest users to not use the same ID for customized Profiles, but I guess we cannot stop them from doing it, I don't know what use cases people have.

You're right. The input content can contain whatever. However I was talking about how to explain users which ID they should use in `oscap` tool. 

> Something I noticed now, is that SCAP Workbench 1.1.5 adds 2 to end of Profile ID when provided identical ID to the one being customized. So users shadowing Profiles may be doing it by hand?

I think yes.

> Suggestion: Needs to specify Profile to apply with --profile.

Thanks, but that doesn't solve the issue completely. It doesn't tell users that they have to use the **ID of customized profile**, not the ID of original profile.

> And can we change description of option --profile of xccdf generate from the help to be like the one in manual page? I find the description in man page much more clear.

Yes, we can!

> I don't understand how to use --tailoring-id. Couldn't figure out from manual or help. Tried following and error message didn't help me.
I think you should use the ID of datastream component, not the ID from Tailoring element. :crying_cat_face: It's terrible. :sob:

Reply:
I don't understand the flow of information for `--tailoring-id`.

It seems that `--tailoring-file` and `--tailoring-id` should specify where to get the customized Profile to apply over the input source data stream.

The DataStream SSG generates for RHEL7 contains only one data-stream component.
So I try using a component from datastream that refers to a Benchmark:
```
[coruscant git]$ oscap xccdf generate fix --output fix.sh --tailoring-id scap_org.open-scap_comp_ssg-rhel7-xccdf-1.2.xml --profile xccdf_org.ssgproject.content_profile_standard --benchmark-id xccdf_org.ssgproject.content_benchmark_RHEL-7 scap-security-guide/build/ssg-rhel7-ds.xml 
OpenSCAP Error: No 'scap_org.open-scap_comp_ssg-rhel7-xccdf-1.2.xml' component ref found in file 'scap-security-guide/build/ssg-rhel7-ds.xml' in datastream of id 'scap_org.open-scap_datastream_from_xccdf_ssg-rhel7-xccdf-1.2.xml'. [ds_sds_session.c:316]
Could not extract scap_org.open-scap_comp_ssg-rhel7-xccdf-1.2.xml with all dependencies from datastream. [ds_sds_session.c:224]
```
Even tried adding a component with tailoring content to the DataStream and using it, no success.
```
[coruscant git]$ oscap xccdf generate fix --output fix.sh --tailoring-id scap_org.open-scap_comp_ssg-rhel7-tailoring-xccdf-1.2.xml --profile xccdf_org.ssgproject.content_profile_standard2 --benchmark-id xccdf_org.ssgproject.content_benchmark_RHEL-7 ssg-rhel7-ds.xml 
OpenSCAP Error: No 'scap_org.open-scap_comp_ssg-rhel7-tailoring-xccdf-1.2.xml' component ref found in file 'ssg-rhel7-ds.xml' in datastream of id 'scap_org.open-scap_datastream_from_xccdf_ssg-rhel7-xccdf-1.2.xml'. [ds_sds_session.c:316]
Could not extract scap_org.open-scap_comp_ssg-rhel7-tailoring-xccdf-1.2.xml with all dependencies from datastream. [ds_sds_session.c:224]
```

Reply:
@yuumasato How did you create your datastream?